modules/ssm/main.tf

```diff
@@ -0,0 +1,7 @@
+resource "aws_ssm_parameter" "secret_parameters" {
+  for_each = var.secrets
+
+  name  = "/${var.env_namespace}/${each.key}"
+  type  = "String"
```

Comment:
Any sensitive values won't be encrypted - wondering if we should encrypt any sensitive values with type `"SecureString"`. Although you haven't set up KMS it should use the default key AWS creates for you - although viewable in AWS Console with anyone with those permissions for the default key.

Some benefits seem to be:
- Not displayed in logs
- Encrypted on disk

Let me know what you think on this one. We will store DB + Elasticache connection details here too? I know we have `secret_key_base` for now.

https://docs.aws.amazon.com/kms/latest/developerguide/services-parameter-store.html
https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html

Reply:
not sure why we need to use the `SecureString` to encrypt the sensitive string in SSM, if that is so, is it better to use KMS over SSM + KMS encrypt key?
In this case, I just want to use the `sensitive` filter from Terraform, it's one-way encrypt and does not reveal data, do you think it's enough for the IC?

Reply:
I think it would be useful for sensitive information such as the `secret_key_base`. As per the link above, it states the following use cases:

```
- You want to use data/parameters across AWS services without exposing the values as plaintext in commands, functions, agent logs, or CloudTrail logs.
- You want to control who has access to sensitive data.
- You want to be able to audit when sensitive data is accessed (CloudTrail).
- You want to encrypt your sensitive data, and you want to bring your own encryption keys to manage access.
```

So, benefits such as controlling access to the values in the AWS console and using IAM Policies, audit logs etc. I think for the IC we can keep it as it is as it follows the `infra` approach we have. I think in real-world scenario we wouldnt be using plaintext here especially for secret-key base when using `parameter store`